support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/web/DefaultOAuth20RequestParameterResolver.java

```diff
@@ -276,7 +275,10 @@ public Set<String> resolveSupportedPromptValues(final String url) {
 
     @Override
     public boolean isParameterOnQueryString(final WebContext context, final String name) {
-        val queryString = ((JEEContext) context).getNativeRequest().getQueryString();
-        return context.getRequestParameter(name).isPresent() && StringUtils.contains(queryString, name + '=');
+        val queryString = context.getQueryString();
+        if (queryString.isPresent()) {
+            return context.getRequestParameter(name).isPresent() && StringUtils.contains(queryString.get(), name + '=');
+        }
```

Comment:
Idea: How about?

```
return context.isQueryStringParameter(name)
```

...and you'd move all this logic over inside the context. (Make this be a default method on the interface).

Reply:
The `WebContext` only intents to propose low-level methods. Though, this logic can be available in the pac4j `WebContextHelper`.

Reply:
This is what I did: https://github.com/apereo/cas/pull/5810/commits/b6e5d020dfa0dc5a30228389d75fc5dfcabfabc2